hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/InsertProcessorSupplier.java

```diff
@@ -130,4 +137,29 @@ private Document rowToDoc(JetSqlRow row) {
         return doc;
     }
 
+    @Override
+    public void writeData(ObjectDataOutput out) throws IOException {
+        out.writeString(connectionString);
+        out.writeString(databaseName);
+        out.writeString(collectionName);
+        out.writeStringArray(paths);
+        out.writeObject(writeMode);
+        out.writeObject(types);
+        out.writeObject(externalTypes);
```

Comment:
Won't these 3rd party object types serialize using Java serialization anyway? Maybe we could use something more efficient here? Or do we handle enums somehow special?

Reply:
True, I've added serialization to array of ints

Reply:
That creates unnecessary objects for both serialization and deserialization.

Just write int as number of items and then all int values for each BsonType.